### PR TITLE
Change wording of long lines to lower restriction

### PR DESCRIPTION
**Pull request by fuelxc**

### CODE REVIEW

**Comment by coreycondardo** [comment]
Stolen and selectively edited from here: http://softwareengineering.stackexchange.com/questions/604/is-the-80-character-limit-still-relevant-in-times-of-widescreen-monitors

Reasons I think we shouldn't accept this PR:

* "Even in the era of widescreen monitors, I like to have several windows open side by side, showing different parts of the code. If you have lines that run over 80 columns, you need to deal with the editor wrapping the lines (which is ugly and makes the code harder to navigate around), or widen you windows such that you can't fit as many on screen at once."

* "Even if you don't usually edit this way, if you ever use a side-by-side diff tool, you will appreciate files with reasonable line lengths which will make your diff easier to view."

* "There's also an issue of code density. I like to have a lot of context when reading code. It is much, much quicker to glance up and down a window than it is to scroll. If you have very long lines, you also tend to have lines that vary in length a lot, leading to a lot of wasted screen real estate and being able to fit less code on screen at a given time overall."

* "And finally, if you have very long lines, then that generally means that you have very complicated lines, deep indendation, or that you have very long identifiers. All of these can be a problem. Complicated lines are probably doing too much; if you can break it down into several simpler lines, you probably should. Deep indentation means that you're probably nesting too many loops and conditionals, which can make your code flow confusing; considering refactoring into several functions. And if your identifiers are too long, it can make reading your code very difficult. People generally recognize words as individual units; they don't read every character one by one, but look at the overall shape of the word. Long identifiers are harder to distinguish this way, and usually if they are that long, they contain redundant or repetitive information."

* Added by me... Github code reviews are a lot easier when the line length is ~80

**Comment by coreycondardo** [comment]
Additionally, I prefer that the style guide is specific and measurable, making `gratuitous or excessively long` too vague in my humble opinion.

To get ahead of the canned response to this is ("_its just a guide!_"), I'll say our first guide line covers that case:
>"You should break these rules only if you have a good reason or your [reviewer] lets you."

Given the current guidelines you have every right to attempt to convince your reviewer that it's appropriate to let a 80+ character line go.

Therefore, I'd ask to see a specific and measurable amount of characters if we are to propose a change to this rule.

File: ruby/README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ Formatting
 
 * Avoid inline comments.
 * Files should end with a newline (unix standard).
-* Break long lines after 80 characters.
+* Break gratuitous or excessively long lines.
```
Comment (semmons99) [inline ruby/README.md]: We try to be much more precise with this guide. What is the definition of "gratuitous" or "excessively" 120 characters, 150, 200?

**Comment by dmccraw** [comment]
I have found that the 80 character limit to be useful in a lot of ways. I am constantly diffing code and to diff two 100 character lines next to each other can be difficult. I also work with 3 windows side by side. (This can be overcome with word wrapping of course.) I think the biggest advantage I have seen is the understandability of the code. By having a chunk of code spanning multiple lines I am able to understand the data transformation much better.
```
    params
      .fetch("custom_field") { {} }
      .map { |id, args|
        parse_search_args(args).map { |arg| {:id => id}.merge(arg) }
      }
      .reduce([]) { |parms, parm| parms.concat(parm) }
```
is way easier to understand than
```
    params.fetch("custom_field") { {} } .map { |id, args| parse_search_args(args).map { |arg| {:id => id}.merge(arg) }}.reduce([]) { |parms, parm| parms.concat(parm) }
```
Strings can often make a line break more than 80 characters.
```
      flash[:custom_error] = "The import file could not be removed because it has been used to create events."
```
vs
```
      flash[:custom_error] = 
        "The import file could not be removed because it has been used to create events."
```
I have found that the second is more readable because usually you care about either the string or the variable. By splitting it into two lines my brain has a much easier time determining at a glance which I care about.

```
    link_to_function(("<i><img src='#{asset_path("#{options[:icon]}.gif")}'/></i><span>#{text}</span>").html_safe, "jQuery('##{options[:off]}').hide();jQuery('##{options[:on]}').show();", :class=>"button #{options[:size]}")
```
vs
```
    link_to_function(
      ("<i><img src='#{asset_path("#{options[:icon]}.gif")}'/></i><span>#{text}</span>").html_safe, 
      "jQuery('##{options[:off]}').hide();jQuery('##{options[:on]}').show();",
      :class=>"button #{options[:size]}"
    )
```

File: ruby/README.md (hunk 1 above)
Comment (fuelxc) [inline ruby/README.md]: That depends on the circumstances.  Setting to 80 no matter what can make code harder to read just as often as it makes it easier.  even in this guide I see 8 places that we go beyond 80 characters.

Comment (dmccraw) [inline ruby/README.md]: But it's an 80 character limit with a bailout:
> "You should break these rules only if you have a good reason or your [reviewer] lets you."
You just need to justify why it should be more than 80 characters.

**Comment by fuelxc** [comment]
I find trying to adhere to 80 characters make code harder to read in some cases.  Readability should be our goal and having a set column limit can make code harder to read just as often as it makes it easier.

You add that to the rule that methods are only allowed 5 lines and classes only 100 lines and things break down in a hurry.

```
class foo
  def bar(message)
    message.tap do |msg|
      well_named = Class.method(value1, "context to the method call", {option: 1})
    end
  end
end
```

That line is over the 80 character limit by 2.  I would argue vehemently that making it under the limit would make it more unreadable.  

```
class foo
  def bar(message)
    message.tap do |msg|
      well_named = Class.method(value1, "context to the method call", 
        {
          option: 1
        }
      )
    end
  end

  # or

  def bar(message)
    message.tap do |msg|
      well_named = Class.method(
        value1, 
        "context to the method call", 
        {
          option: 1
        }
      )
    end
  end
end
```

that also puts it over our 5 line limit as does this one:

```
    params
      .fetch("custom_field") { {} }
      .map { |id, args|
        parse_search_args(args).map { |arg| {:id => id}.merge(arg) }
      }
      .reduce([]) { |parms, parm| parms.concat(parm) }
```
(though I do believe that is more readable)

Putting the onus on the committer to make a compelling argument to exceed the arbitrary 80 char limit seems counter productive.

I would error on the side of trust a developer to make it readable and if it isn't, then kick it back.

File: ruby/README.md (hunk 1 above)
Comment (fuelxc) [inline ruby/README.md]: See comment below with my opinion(s) :)

**Comment by glanotte** [comment]
I like the 80 character limit as well. It forces me to think about things more which I like. I generally am not a fan of ambiguity in a style guide. The purpose of a style guide is to be that, if we are going to be ambiguous, we can just throw this away.

**Comment by coreycondardo** [comment]
```
      well_named = Class.method(value1, "context to the method call", {option: 1})
```
It may be contrived, but if I saw this in a PR I'd question if we should be accepting 3 unnamed arguments to a method. I'd prefer either named arguments with this many (which will make the line even longer) or some kind of an options hash (which will make it much shorter). This is the beauty of the 80 line limit. It forces you to think about why it is 80+ in the first place, and if there is a better way to shorten it. Typically shortening isn't just breaking it onto multiple lines, but making the thing that is long better.

File: ruby/README.md (hunk 1 above)
Comment (coreycondardo) [inline ruby/README.md]: By that logic, can't we just throw away the guide and say "do the right thing?" I like having hard and fast rules, they give me good structure.

**Comment by fuelxc** [comment]
Almost every Rails view helper has that format (maybe not the `Class` call)

**Comment by randy-girard** [comment]
We are slowly moving away from rails and shouldn't let their style influence ours.

**Comment by fuelxc** [comment]
that was just an example.  Redis gem is also another that uses that convention.  It is a common convention for 1 or 2 positional args with an options hash.

**Comment by glanotte** [comment]
Not a fan of this PR, if you want to see what happens without a style guide and allowing developers to use their better judgement, you can look at classic. I have been working within the 80 character limit for 3 years now and have yet to run into any compelling reason to change the guideline.

**Pull request closed without merging**